components/pkg-export-kubernetes/src/manifest.rs

```diff
@@ -129,7 +129,7 @@ impl Manifest {
 
     /// Generates the manifest as a string and writes it to `write`.
     pub fn generate(&mut self, write: &mut Write) -> Result<()> {
-        let out = ManifestJson::new(&self).into_string()?;
+        let out: String = ManifestJson::new(&self).into();
```

Comment:
I don't think you decide to use Into trait or not base weather not a paricular usage of it will have to use type annotation. if the usage of 'out' afterwards would have made the type obvious to the compiler, we wouldn't need to annotate.

I did indeed go with implementing Display train actually but it felt wrong since that's mostly to create a user-visible string representation to show to user in messages and stuff. Although at that time, I hadn't decided that this method shouldn't return a Result. Now that I think of it, ToString() might be better indeed cause it's best to be explicit if possible but the docs discouraging implementations of it, makes me a bit vary of this trait.

Reply:
Ask on IRC :)